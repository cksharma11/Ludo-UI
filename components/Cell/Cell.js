import React, { useEffect, useState } from 'react';
import PropTypes from 'prop-types';
import CellStyles from './Cell.style';
import Coin from '../Coin/Coin';

const Cell = ({ id, className, gameData, playerId }) => {
  const [placedCoins, setPlacedCoins] = useState([]);

  useEffect(() => {
    const newPlacedCoins = gameData.coinsPosition.filter(
      (coin) => coin.position === +id
    );
    setPlacedCoins(newPlacedCoins);
  }, [gameData]);

  return (
    <div id={id} className={className}>
      {placedCoins.length
        ? placedCoins.map((coin) => (
            <Coin
              color={coin.color}
              id={coin.id}
              gameData={gameData}
              playerId={playerId}
            />
          ))
        : null}
      <style jsx>{CellStyles}</style>
    </div>
  );
};

Cell.defaultProps = {
  id: 0,
  className: ''
};

Cell.propTypes = {
  id: PropTypes.number,
  className: PropTypes.string,
  gameData: PropTypes.object.isRequired,
  playerId: PropTypes.number.isRequired
};

export default Cell;
