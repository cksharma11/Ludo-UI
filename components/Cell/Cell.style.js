import css from 'styled-jsx/css';

const CellStyles = css`
  .cell {
    display: flex;
    width: 40px;
    height: 40px;
    justify-content: center;
    border: 1px solid #433e5d;
  }

  .red {
    background: rgb(231, 96, 96);
  }

  .green {
    background: rgb(112, 236, 112);
  }

  .yellow {
    background: rgb(230, 230, 91);
  }

  .blue {
    background: rgb(84, 84, 235);
  }

  .dark_red {
    background: red;
  }

  .dark_green {
    background: green;
  }

  .dark_yellow {
    background: yellow;
  }

  .dark_blue {
    background: blue;
  }

  .safe-cell::before {
    font-weight: bold;
    color: navy;
    font-size: 15px;
    content: 'x';
    display: flex;
    justify-content: center;
    align-items: center;
  }
`;

export default CellStyles;
