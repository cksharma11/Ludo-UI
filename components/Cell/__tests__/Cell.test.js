import React from 'react';
import { shallow } from 'enzyme';
import Cell from '../Cell';

describe('Cell', () => {
  it('should render properly', () => {
    const wrapper = shallow(<Cell />);
    expect(wrapper.debug()).toMatchSnapshot();
  });
});
