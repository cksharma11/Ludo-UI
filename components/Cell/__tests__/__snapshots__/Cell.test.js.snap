// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Cell should render properly 1`] = `
"<div id={0} className=\\"jsx-613345661 \\">
  <JSXStyle id=\\"613345661\\">
    .
    c
    e
    l
    l
    .
    j
    s
    x
    -
    6
    1
    3
    3
    4
    5
    6
    6
    1
    {
    d
    i
    s
    p
    l
    a
    y
    :
    -
    w
    e
    b
    k
    i
    t
    -
    b
    o
    x
    ;
    d
    i
    s
    p
    l
    a
    y
    :
    -
    w
    e
    b
    k
    i
    t
    -
    f
    l
    e
    x
    ;
    d
    i
    s
    p
    l
    a
    y
    :
    -
    m
    s
    -
    f
    l
    e
    x
    b
    o
    x
    ;
    d
    i
    s
    p
    l
    a
    y
    :
    f
    l
    e
    x
    ;
    w
    i
    d
    t
    h
    :
    4
    0
    p
    x
    ;
    h
    e
    i
    g
    h
    t
    :
    4
    0
    p
    x
    ;
    -
    w
    e
    b
    k
    i
    t
    -
    b
    o
    x
    -
    p
    a
    c
    k
    :
    c
    e
    n
    t
    e
    r
    ;
    -
    w
    e
    b
    k
    i
    t
    -
    j
    u
    s
    t
    i
    f
    y
    -
    c
    o
    n
    t
    e
    n
    t
    :
    c
    e
    n
    t
    e
    r
    ;
    -
    m
    s
    -
    f
    l
    e
    x
    -
    p
    a
    c
    k
    :
    c
    e
    n
    t
    e
    r
    ;
    j
    u
    s
    t
    i
    f
    y
    -
    c
    o
    n
    t
    e
    n
    t
    :
    c
    e
    n
    t
    e
    r
    ;
    b
    o
    r
    d
    e
    r
    :
    1
    p
    x
     
    s
    o
    l
    i
    d
     
    #
    4
    3
    3
    e
    5
    d
    ;
    }
    .
    r
    e
    d
    .
    j
    s
    x
    -
    6
    1
    3
    3
    4
    5
    6
    6
    1
    {
    b
    a
    c
    k
    g
    r
    o
    u
    n
    d
    :
    r
    g
    b
    (
    2
    3
    1
    ,
    9
    6
    ,
    9
    6
    )
    ;
    }
    .
    g
    r
    e
    e
    n
    .
    j
    s
    x
    -
    6
    1
    3
    3
    4
    5
    6
    6
    1
    {
    b
    a
    c
    k
    g
    r
    o
    u
    n
    d
    :
    r
    g
    b
    (
    1
    1
    2
    ,
    2
    3
    6
    ,
    1
    1
    2
    )
    ;
    }
    .
    y
    e
    l
    l
    o
    w
    .
    j
    s
    x
    -
    6
    1
    3
    3
    4
    5
    6
    6
    1
    {
    b
    a
    c
    k
    g
    r
    o
    u
    n
    d
    :
    r
    g
    b
    (
    2
    3
    0
    ,
    2
    3
    0
    ,
    9
    1
    )
    ;
    }
    .
    b
    l
    u
    e
    .
    j
    s
    x
    -
    6
    1
    3
    3
    4
    5
    6
    6
    1
    {
    b
    a
    c
    k
    g
    r
    o
    u
    n
    d
    :
    r
    g
    b
    (
    8
    4
    ,
    8
    4
    ,
    2
    3
    5
    )
    ;
    }
    .
    d
    a
    r
    k
    _
    r
    e
    d
    .
    j
    s
    x
    -
    6
    1
    3
    3
    4
    5
    6
    6
    1
    {
    b
    a
    c
    k
    g
    r
    o
    u
    n
    d
    :
    r
    e
    d
    ;
    }
    .
    d
    a
    r
    k
    _
    g
    r
    e
    e
    n
    .
    j
    s
    x
    -
    6
    1
    3
    3
    4
    5
    6
    6
    1
    {
    b
    a
    c
    k
    g
    r
    o
    u
    n
    d
    :
    g
    r
    e
    e
    n
    ;
    }
    .
    d
    a
    r
    k
    _
    y
    e
    l
    l
    o
    w
    .
    j
    s
    x
    -
    6
    1
    3
    3
    4
    5
    6
    6
    1
    {
    b
    a
    c
    k
    g
    r
    o
    u
    n
    d
    :
    y
    e
    l
    l
    o
    w
    ;
    }
    .
    d
    a
    r
    k
    _
    b
    l
    u
    e
    .
    j
    s
    x
    -
    6
    1
    3
    3
    4
    5
    6
    6
    1
    {
    b
    a
    c
    k
    g
    r
    o
    u
    n
    d
    :
    b
    l
    u
    e
    ;
    }
    .
    s
    a
    f
    e
    -
    c
    e
    l
    l
    .
    j
    s
    x
    -
    6
    1
    3
    3
    4
    5
    6
    6
    1
    :
    :
    b
    e
    f
    o
    r
    e
    {
    f
    o
    n
    t
    -
    w
    e
    i
    g
    h
    t
    :
    b
    o
    l
    d
    ;
    c
    o
    l
    o
    r
    :
    n
    a
    v
    y
    ;
    f
    o
    n
    t
    -
    s
    i
    z
    e
    :
    1
    5
    p
    x
    ;
    c
    o
    n
    t
    e
    n
    t
    :
    &#39;
    x
    &#39;
    ;
    d
    i
    s
    p
    l
    a
    y
    :
    -
    w
    e
    b
    k
    i
    t
    -
    b
    o
    x
    ;
    d
    i
    s
    p
    l
    a
    y
    :
    -
    w
    e
    b
    k
    i
    t
    -
    f
    l
    e
    x
    ;
    d
    i
    s
    p
    l
    a
    y
    :
    -
    m
    s
    -
    f
    l
    e
    x
    b
    o
    x
    ;
    d
    i
    s
    p
    l
    a
    y
    :
    f
    l
    e
    x
    ;
    -
    w
    e
    b
    k
    i
    t
    -
    b
    o
    x
    -
    p
    a
    c
    k
    :
    c
    e
    n
    t
    e
    r
    ;
    -
    w
    e
    b
    k
    i
    t
    -
    j
    u
    s
    t
    i
    f
    y
    -
    c
    o
    n
    t
    e
    n
    t
    :
    c
    e
    n
    t
    e
    r
    ;
    -
    m
    s
    -
    f
    l
    e
    x
    -
    p
    a
    c
    k
    :
    c
    e
    n
    t
    e
    r
    ;
    j
    u
    s
    t
    i
    f
    y
    -
    c
    o
    n
    t
    e
    n
    t
    :
    c
    e
    n
    t
    e
    r
    ;
    -
    w
    e
    b
    k
    i
    t
    -
    a
    l
    i
    g
    n
    -
    i
    t
    e
    m
    s
    :
    c
    e
    n
    t
    e
    r
    ;
    -
    w
    e
    b
    k
    i
    t
    -
    b
    o
    x
    -
    a
    l
    i
    g
    n
    :
    c
    e
    n
    t
    e
    r
    ;
    -
    m
    s
    -
    f
    l
    e
    x
    -
    a
    l
    i
    g
    n
    :
    c
    e
    n
    t
    e
    r
    ;
    a
    l
    i
    g
    n
    -
    i
    t
    e
    m
    s
    :
    c
    e
    n
    t
    e
    r
    ;
    }
  </JSXStyle>
</div>"
`;
